cpp/dolfinx/fem/assemble_matrix_impl.h

```diff
@@ -185,10 +186,11 @@ void assemble_cells(
 /// mesh
 /// @param cell_info1 The cell permutation information for the trial function
 /// mesh
+/// @param get_perm Function to get the facet permutations
```

Comment:
Make the docstring clearer. 

I'm not sure we want to pass a `std::function` that does very little work - there can be an overhead.